### PR TITLE
feat(superuser modal): Removed the password and u2f check for superuser modal FE 

### PR DESCRIPTION
**Pull request by maxiuyuan**
Removed the password and u2f check on superuser modal for non on-prem users. As discussed with @RyanSkonnord and @leedongwei, this check provides nothing of value.

BE here: https://github.com/getsentry/sentry/pull/33663

### CODE REVIEW

**Comment by RyanSkonnord** [comment]
Just making sure, this affects only the superuser check for platform administrators, and not the so-called "sudo" modal that interrupts regular users on higher-risk operations?

**Comment by maxiuyuan** [comment]
> Just making sure, this affects only the superuser check for platform administrators, and not the so-called "sudo" modal that interrupts regular users on higher-risk operations?

Yup, For the sudo modal, they'll still be required their password/u2f check

File: static/app/components/modals/sudoModal.tsx [hunk 1]
```diff
@@ -177,7 +158,7 @@ class SudoModal extends React.Component<Props, State> {
               initialData={{isSuperuserModal: isSuperuser}}
               resetOnError
             >
-              {!isSelfHosted && showAccessForms && isSuperuser && (
+              {!isSelfHosted && isSuperuser && (
```
Comment (leedongwei) [inline static/app/components/modals/sudoModal.tsx:161]: This `isSuperuser` is not needed because it's already checked in Line 151.

**Review by leedongwei — APPROVED** [review]

File: static/app/components/modals/sudoModal.tsx (hunk 1 above)
Comment (nhsiehgit) [reply static/app/components/modals/sudoModal.tsx:161]: ^double this.

**Review by nhsiehgit — APPROVED** [review]
+1 pending comment

**Review by RyanSkonnord — APPROVED** [review]

**Pull request merged**